### PR TITLE
feat: created cart management functionality

### PR DESCRIPTION
**Pull request by arjunbharti**
Created cart management:

- Add products to cart
- Delete products from cart
- Increment or decrement the product in the cart
- Get updated total number of products count
- Get updated total amount of products

https://user-images.githubusercontent.com/60930192/159943099-86c622ad-c246-4e55-9a04-a9713f220014.mp4

### CODE REVIEW

File: src/contexts/cart-context.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+import { createContext, useContext, useReducer } from 'react'
+import { cartReducer } from '../reducers/cart-reducer';
+const CartContext = createContext(null);
+const useCart = () => useContext(CartContext);
+
+const CartProvider = ({ children }) => {
+    const [cartItems, setCartItems] = useReducer(cartReducer, {
```
Comment (jeetbhalani15) [inline src/contexts/cart-context.js:7]: just a suggestion u can use cartDispatch if you are using useReducer to make it more readable & meaningful.

**Review by jeetbhalani15 — APPROVED** [review]
-Added some suggestions
- you can also show no item in cart if there is no cartitems & hide cartbilling.
- -rest looks good

**Review by sahiba0915 — COMMENTED** [review]
code looks clean and good to merge

File: src/contexts/cart-context.js (hunk 1 above)
Comment (arjunbharti) [reply src/contexts/cart-context.js:7]: this naming was more convenient to me, so i use it. Although, i will keep this suggestion in mind

**Pull request merged**